crates/eww/src/widgets/widget_definitions.rs

```diff
@@ -536,10 +536,55 @@ fn build_gtk_image(bargs: &mut BuilderArgs) -> Result<gtk::Image> {
                 gtk_widget.set_from_animation(&pixbuf_animation);
+            } else if path.ends_with(".svg") && !fill.is_empty() {
+                let svg_data = std::fs::read_to_string(std::path::PathBuf::from(path.clone()))?;
+
+                // The fastest way to add/change fill color
+                let svg_data = if svg_data.contains("fill=") {
+                    let reg = regex::Regex::new(r#"fill="[^"]*""#)?;
+                    reg.replace(&svg_data, &format!("fill=\"{}\"", fill))
+                } else {
+                    let reg = regex::Regex::new(r"<svg")?;
+                    reg.replace(&svg_data, &format!("<svg fill=\"{}\"", fill))
+                };
+
+                let pixbuf_svg = gtk::gdk_pixbuf::PixbufLoader::with_type("svg")?;
+                // To determine the size of svg image
+                // NOTE: Convertions to f64 are intentional to prevent negative or zero values
+                // which cause some images with rectangular dimensions to not be displayed
+                // if at least one of the image-width/image-height attributes is used
+                //
+                // For example, we have a rectangular image with dimensions 300x130(width = 300, height = 130),
+                // and the user only wants to use the attribute :image-width 18,
+                // then the aspect ratio to calculate the height would be
+                // aspect_ratio_h = (height / width); // return 0, then aspect_ration_h * 18 = 0
+                // aspect_ratio_h = (height as f64 / width as f64); // return 0.4333333333333, then aspect_ratio_h * 18.0 = 7.8 as i32 = 7
+                pixbuf_svg.connect_size_prepared(move |pixbuf_svg, svg_width, svg_height| {
+                    let aspect_ratio_h = svg_height as f64 / svg_width as f64;
+                    let aspect_ratio_w = svg_width as f64/ svg_height as f64;
+
+                    match (image_width, image_height) {
+                        // If only define the image-width attribute, then preserve the aspect ratio for the svg height
+                        (w, _) if w >= 0 => pixbuf_svg.set_size(w, (aspect_ratio_h * w as f64) as i32),
+                        // If only define the image-height attribute, then preserve the aspect ratio for the svg width
+                        (_, h) if h >= 0 => pixbuf_svg.set_size((aspect_ratio_w * h as f64) as i32, h),
+                        // If both attributes image-width and image-height are difined, then use it
+                        (w, h) if w >= 0 && h >= 0 => pixbuf_svg.set_size(w, h),
+                        _ => {}
+                    };
+                });
+                pixbuf_svg.write(svg_data.as_bytes())?;
+                pixbuf_svg.close()?;
+
+                gtk_widget.set_from_pixbuf(pixbuf_svg.pixbuf().as_ref());
             } else {
                 let pixbuf = gtk::gdk_pixbuf::Pixbuf::from_file_at_size(std::path::PathBuf::from(path), image_width, image_height)?;
```

Comment:
```suggestion
            } else {
                let pixbuf;
                // populate the pixel buffer
                if path.ends_with(".svg") && !fill.is_empty() {
                    let svg_data = std::fs::read_to_string(std::path::PathBuf::from(path.clone()))?;
                    // The fastest way to add/change fill color
                    let svg_data = if svg_data.contains("fill=") {
                        let reg = regex::Regex::new(r#"fill="[^"]*""#)?;
                        reg.replace(&svg_data, &format!("fill=\"{}\"", fill))
                    } else {
                        let reg = regex::Regex::new(r"<svg")?;
                        reg.replace(&svg_data, &format!("<svg fill=\"{}\"", fill))
                    };
                    let stream = gtk::gio::MemoryInputStream::from_bytes(&gtk::glib::Bytes::from(svg_data.as_bytes()));
                    pixbuf = gtk::gdk_pixbuf::Pixbuf::from_stream_at_scale(&stream, image_width, image_height, true, None::<&gtk::gio::Cancellable>)?;
                    stream.close(None::<&gtk::gio::Cancellable>)?;
                } else {
                    pixbuf = gtk::gdk_pixbuf::Pixbuf::from_file_at_size(std::path::PathBuf::from(path), image_width, image_height)?;
                }

```

Reply:
Nice implementation, 100x better than mine and simpler

Reply:
I think it would be best to close this PR and for you to submit a new one, since you did much better and the truth is that I just wanted to give an idea since I am very new to this.

Reply:
i can open a new pr if you want, but give yourself some credit too, i don't know anything about the regex stuff you're doing there and the idea to add this never crossed my mind ^^'
i'll also look into the aspect ratio attribute functionality since both concern the same widget so give me a moment to evaluate that

Reply:
The aspect ratio was just to simulate what Pixbuf::from_file_at_size already does, since set_size from PixbufLoader was only taken if both values ​​were greater than 0, if it was only one of the two it was ignored.